From 3945e1cda7e2de61646f3a9133ec9fa296eb73cb Mon Sep 17 00:00:00 2001
From: Raul E Rangel <rrangel@chromium.org>
Date: Thu, 21 Mar 2024 10:41:38 -0600
Subject: [PATCH 1/8] FROMLIST: BACKPORT: save-build-env: Strip out MAKEOPTS
 and NINJAOPTS

MAKEOPTS and NINJAOPTS contains a host specific value. If the same
package is built on a host with different core counts, the binpkgs will
have different hashes. This change drops MAKEOPTS after src_install so
that it's not included in final environment.gz. This change also adds
them to the special_env_vars.py so that we always have the most up to
date version of these variables.

Bug: https://bugs.gentoo.org/914441
Bug: https://bugs.gentoo.org/736794
Signed-off-by: Raul E Rangel <rrangel@chromium.org>

(am from https://github.com/gentoo/portage/pull/1311/commits/4715ed37726207a423b0a2dcfa625b636fb98b66)

BUG=b:281732865, b:266973461
TEST=bazel build \@portage//internal/packages/stage1/target/host/chromiumos/sys-libs/glibc:2.37-r7_hermetic_test

Change-Id: I20a1b5007d3e5f3074cb5515c3ef748e8910f9bd
---
 bin/save-ebuild-env.sh                                 | 6 ++++++
 lib/portage/package/ebuild/_config/special_env_vars.py | 3 ++-
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/usr/lib/portage/PYTHON_VERSION/save-ebuild-env.sh b/usr/lib/portage/PYTHON_VERSION/save-ebuild-env.sh
index cd8602ae0c..c79a59b42b 100644
--- a/usr/lib/portage/PYTHON_VERSION/save-ebuild-env.sh
+++ b/usr/lib/portage/PYTHON_VERSION/save-ebuild-env.sh
@@ -24,6 +24,12 @@ __save_ebuild_env() {
 				unset PYTHONPATH
 			fi
 		fi
+
+		# These variables contains build host specific configuration. We
+		# want binpkgs generated on different sized hosts to be
+		# identical, so strip them from the binpkg. It's also not needed
+		# for installing / removing a package.
+		unset MAKEOPTS NINJAOPTS
 	fi
 
 	# misc variables inherited from the calling environment
diff --git a/usr/PYTHON_LIBDIR/PYTHON_VERSION/site-packages/portage/package/ebuild/_config/special_env_vars.py b/usr/PYTHON_LIBDIR/PYTHON_VERSION/site-packages/portage/package/ebuild/_config/special_env_vars.py
index 7c0971b245..1a5960115b 100644
--- a/usr/PYTHON_LIBDIR/PYTHON_VERSION/site-packages/portage/package/ebuild/_config/special_env_vars.py
+++ b/usr/PYTHON_LIBDIR/PYTHON_VERSION/site-packages/portage/package/ebuild/_config/special_env_vars.py
@@ -48,7 +48,8 @@ environ_whitelist += [
 	"EBUILD_FORCE_TEST",
 	"EBUILD_PHASE", "EBUILD_PHASE_FUNC", "ECLASSDIR", "ECLASS_DEPTH", "ED",
 	"EMERGE_FROM", "EPREFIX", "EROOT", "ESYSROOT",
-	"FEATURES", "FILESDIR", "HOME", "MERGE_TYPE", "NOCOLOR", "PATH",
+	"FEATURES", "FILESDIR", "HOME",
+	"MAKEOPTS", "MERGE_TYPE", "NINJAOPTS", "NOCOLOR", "PATH",
 	"PKGDIR",
 	"PKGUSE", "PKG_LOGDIR", "PKG_TMPDIR",
 	"PORTAGE_ACTUAL_DISTDIR", "PORTAGE_ARCHLIST", "PORTAGE_BASHRC_FILES",
-- 
2.44.0.769.g3c40516874-goog

